app/policies/lesson_policy.rb

```diff
@@ -1,14 +1,18 @@
 class LessonPolicy < ApplicationPolicy
   def create_lesson?
-    user.admin? || (user.instructor? && record.courses.where(id: user.id).present?)
+    ( user.enrollments.where(course_id: record.course_id, enrollment_type: "instructor" ).present?)
```

Comment:
can't we create lesson with admin?

Reply:
When a course is created by a person, he or she is the instructor of the course. In this case, their enrollment type is instructor. If an admin creates a course, then he or she can add lessons for this course. but I didn't give any permission to add a lesson to another person's course. But as an admin, he can delete any course or any lesson if anything goes wrong.
Thank you.